scripts/validate-naming.py

```
@@ -96,8 +109,18 @@ def docs_with_kind(path):
             out.append((api or "", kind))
     return out
 
+def count_docs(path):
+    """Count YAML documents with any non-comment content (kind-less included)."""
+    with open(path, encoding="utf-8", errors="replace") as f:
+        text = f.read()
+    return sum(
+        1 for chunk in re.split(r"(?m)^---[ \t]*$", text)
+        if any(l.strip() and not l.lstrip().startswith("#") for l in chunk.splitlines())
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)